File: src/07.r
library(tidyverse)

input <- read_lines("data/07.txt")

sizes <- integer()
stack <- integer()

for (line in input) {
  if (grepl("^\\d", line)) stack <- stack + as.integer(str_extract(line, "\\d+"))
  if (line == "$ ls") stack <- append(stack, 0)
  if (line == "$ cd ..") {
    sizes <- append(sizes, tail(stack, 1))
    stack <- head(stack, -1)
  }
}

sizes <- append(sizes, stack)

sum(sizes[sizes < 100000])
min(sizes[sizes > max(sizes) - 40000000])

File: src/07.py
input = open('./data/07.txt', 'r').read().splitlines()

sizes = []
stack = []

for line in input:
  match line.split():
    case ["$", "ls"]:
      stack.append(0)
    case ["$", "cd", ".."]:
      sizes.append(stack.pop())
    case [size, name] if size.isnumeric():
      stack = list(map(lambda sum: sum + int(size), stack))

sizes = stack + sizes

sum(filter(lambda size: size <= 100_000, sizes))
min(filter(lambda size: size >= max(sizes) - 40_000_000, sizes))
